core/src/main/java/io/aiven/klaw/service/SchemaRegstryControllerService.java

```diff
@@ -270,6 +276,82 @@ public ApiResponse execSchemaRequestsDecline(String avroSchemaId, String reasonF
+    }
+
+    Integer userTeamId = commonUtilsService.getTeamId(userDetails);
+    int tenantId = commonUtilsService.getTenantId(getUserName());
+    int teamId = commonUtilsService.getTeamId(userDetails);
+    SchemaRequestModel schemaRequest =
+        buildSchemaRequestFromPromotionRequest(schemaPromotion, teamId, tenantId);
+
+    Optional<Env> optionalEnv = getSchemaEnvFromId(schemaPromotion.getSourceEnvironment());
+
+    if (!optionalEnv.isPresent()) {
+      return ApiResponse.builder()
+          .status(HttpStatus.BAD_REQUEST)
+          .result("Unable to find or access the source Schema Registry")
+          .build();
+    }
+    Env schemaEnv = optionalEnv.get();
+    SortedMap<Integer, Map<String, Object>> schemaObjects =
+        getSchemasFromTopicName(schemaPromotion.getTopicName(), tenantId, schemaEnv);
+    log.info(
+        "getSchemaVersion {}, schemaObjects keySet {}",
+        schemaPromotion.getSchemaVersion(),
+        schemaObjects.keySet());
+    Map<String, Object> schemaObject =
+        schemaObjects.get(Integer.valueOf(schemaPromotion.getSchemaVersion()));
+    // Pretty Print the Json String so that it can be seen clearly in the UI.
+    schemaRequest.setSchemafull(prettyPrintUglyJsonString((String) schemaObject.get("schema")));
+    return uploadSchema(schemaRequest);
+  }
+
+  private SortedMap<Integer, Map<String, Object>> getSchemasFromTopicName(
+      String topicName, int tenantId, Env schemaEnv) throws Exception {
+    KwClusters kwClusters =
+        manageDatabase
+            .getClusters(KafkaClustersType.SCHEMA_REGISTRY, tenantId)
+            .get(schemaEnv.getClusterId());
+
+    SortedMap<Integer, Map<String, Object>> schemaObjects =
+        clusterApiService.getAvroSchema(
+            kwClusters.getBootstrapServers(),
+            kwClusters.getProtocol(),
+            kwClusters.getClusterName() + kwClusters.getClusterId(),
+            topicName,
+            tenantId);
+
+    return schemaObjects;
+  }
+
+  private Optional<Env> getSchemaEnvFromId(String envId) {
+    return manageDatabase
+        .getSchemaRegEnvList(commonUtilsService.getTenantId(getUserName()))
+        .stream()
+        .filter(env -> env.getId().equals(envId))
+        .findFirst();
+  }
+
+  private SchemaRequestModel buildSchemaRequestFromPromotionRequest(
+      SchemaPromotion schemaPromotion, int teamId, int tenantId) {
+    SchemaRequestModel schemaRequest = new SchemaRequestModel();
+    // setup schema Request
+    schemaRequest.setAppname(schemaPromotion.getAppName());
+    schemaRequest.setRemarks(schemaPromotion.getRemarks());
+    schemaRequest.setTeamname(manageDatabase.getTeamNameFromTeamId(tenantId, teamId));
+    schemaRequest.setEnvironment(schemaPromotion.getTargetEnvironment());
+    schemaRequest.setSchemaversion(schemaPromotion.getSchemaVersion());
+    schemaRequest.setTopicname(schemaPromotion.getTopicName());
+    schemaRequest.setForceRegister(schemaPromotion.isForceRegister());
```

Comment:
we don't have this field in SchemaRequest class yet I guess. so a new column in db would be coming later.